src/Mvc/Mvc.ApiExplorer/test/EndpointMetadataApiDescriptionProviderTest.cs

```diff
@@ -384,34 +376,6 @@ public void TestParameterIsRequired()
             Assert.False(barParam.IsRequired);
         }
 
-#nullable enable
-
-        [Fact]
-        public void TestIsRequiredFromBody()
-        {
-            var apiDescription0 = GetApiDescription(([FromBody(EmptyBodyBehavior = EmptyBodyBehavior.Allow)] InferredJsonClass fromBody) => { });
-            var apiDescription1 = GetApiDescription((InferredJsonClass? fromBody) => { });
-            Assert.Equal(1, apiDescription0.ParameterDescriptions.Count);
-            Assert.Equal(1, apiDescription1.ParameterDescriptions.Count);
-
-            var fromBodyParam0 = apiDescription0.ParameterDescriptions[0];
-            Assert.Equal(typeof(InferredJsonClass), fromBodyParam0.Type);
-            Assert.Equal(typeof(InferredJsonClass), fromBodyParam0.ModelMetadata.ModelType);
-            Assert.Equal(BindingSource.Body, fromBodyParam0.Source);
-            Assert.False(fromBodyParam0.IsRequired);
-
-            var fromBodyParam1 = apiDescription1.ParameterDescriptions[0];
-            Assert.Equal(typeof(InferredJsonClass), fromBodyParam1.Type);
-            Assert.Equal(typeof(InferredJsonClass), fromBodyParam1.ModelMetadata.ModelType);
-            Assert.Equal(BindingSource.Body, fromBodyParam1.Source);
-            Assert.False(fromBodyParam1.IsRequired);
-        }
-
-        // This is necessary for TestIsRequiredFromBody to pass until https://github.com/dotnet/roslyn/issues/55254 is resolved.
-        private object RandomMethod() => throw new NotImplementedException();
-
-#nullable disable
-
```

Comment:
@davidfowl FYI, I remove this test as well since we are now skipping creating the api parameter description from body and only getting the value from the metadata. I, also adjusted some other tests.

Reply:
Can we reimplement this test by calling `MapPost` on the `TestEndpointRouteBuilder` and verify the optionality still flows correctly from the `RequestDelegateFactory` to the `EndpointMetadataApiDescriptionProvider` via `IAcceptsMetadta`?

I suspect

Reply:
yes added tests.